tracer/src/Datadog.Trace/Util/ThrowHelper.cs

```diff
@@ -0,0 +1,71 @@
+// <copyright file="ThrowHelper.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+
+namespace Datadog.Trace.Util
+{
+    internal class ThrowHelper
+    {
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        [DebuggerHidden]
+        [DoesNotReturn]
+        internal static void ThrowArgumentNullException(string paramName) => throw new ArgumentNullException(paramName);
```

Comment:
[Why no `[CallerArgumentExpression]` 😢 ](https://andrewlock.net/exploring-dotnet-6-part-11-callerargumentexpression-and-throw-helpers/)

Reply:
Being handled in another PR by lucas. 😊